### ISSUE TITLE
Bug using container queries with angular

### ISSUE THREAD

**Issue by Eoreata** [tailwindlabs/tailwindcss-container-queries#4]
### What version of @tailwindcss/container-queries are you using?

v0.1.0

### What version of Node.js are you using?

18.12.1

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### Reproduction repository

https://github.com/Eoreata/tailwindcss-container-queries-bug

### Describe your issue

The issue is difficult to describe and the bug extremely unpredictable, so I had to make a video of it : https://youtu.be/b5HytyqPjvU

In short, when using container queries with angular, it may (or may not) work and it's hard to say how to reproduce the error. It seems to be related to the use of [class]. Sometimes it works, sometimes it only works after some weird manipulations (like adding commented codes containing container queries).

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

The problem is that you are using dynamically build classes which is not supported: https://tailwindcss.com/docs/content-configuration#dynamic-class-names

You have to remember that Tailwind is a **build** tool, **not a runtime tool**. What it does is it will look at each file in your `content` path as raw text. Then we scan for potential tailwind classes like `bg-red-500` and if we see those, we will generate the CSS for that. There is no runtime, the final product is a simple css file. What you are doing is, you are creating classNames at runtime, and yes they will be correct in your DOM Tree in devtools / your browser, but Tailwind isn't running at that point and since we don't have a runtime, we can't just magically generate new classes. 

If you can search for the literal string `text-black` in your editor, and you see results, then Tailwind will also be able to see that.
If you have this:
```js
let color = 'black'
let klass = `text-${color}`
```

Now if you look for the literal string text-black, your editor won't show results, therefore Tailwind won't see it either.
Tailwind has no idea what "js" is, we don't "run" your code (which makes it super powerful because people use it in HTML and JS, but also in PHP and Rust even).

But how do you solve it? Most recommended to least recommended:
1. don't use dynamically generated classNames for these use cases
2. if you are setting a backgroundImage for example, use a simple style prop instead
3. use custom css properties <div class="text-[color:var(--my-text-color)]" style={{ '--my-text-color': myDynamicVariableValue }}></div>
4. build up a lookup table of classNames based on some information (See 3rd code block starting here https://tailwindcss.com/docs/content-configuration#class-detection-in-depth)
5. safelist classes, this means that all of the safelisted classes will always be generated regardless wether they are used or not. (https://tailwindcss.com/docs/content-configuration#safelisting-classes)

"But sometimes the dynamic classes work", I hear you say, yes, that's because of this:
```js
// File A
<div class="text-black"></div>

// File B — This now works because `text-black` was already generated because of File A
let color = 'black'
let klass = `text-${color}` 
```

So for your exact usecase, you can either safelist them, or build a little lookup table where you list all the classNames based on some variable.

```js
let lookup = {
  xs: `@sx:block`,
  sm: `@sm:block,
  // ...
}

// ...
`[class]="lookup[containerSize]"`
```

Hope this helps!
